astropy/time/tests/test_basic.py

```diff
@@ -999,3 +999,47 @@ def test_isiterable():
     t2 = Time(['1999-01-01 00:00:00.123456789', '2010-01-01 00:00:00'],
               format='iso', scale='utc')
     assert isiterable(t2)
+
+def test_to_datetime():
+    class TimezoneInfo(tzinfo):
+        """
+        This defines a timezone with UTC offset ``utcoffset``, meant as
+        a replacement for the `datetime.tzinfo` objects output by
+        `pytz.timezone`. For example, to simulate a timezone like
+        `pytz.timezone("US/Hawaii")`, try
+        `TimezoneInfo(utcoffset=datetime.timedelta(hours=-10))`
+        """
+        def __init__(self, utcoffset=None, tzname=None, dst=None):
+            self._utcoffset = utcoffset
+            self._tzname = tzname
+            self._dst = dst
+
+        def utcoffset(self, dt):
+            return self._utcoffset
+
+        def tzname(self, dt):
+            return str(self._tzname)
+
+        def dst(self, dt):
+            if self._dst is None:
+                return timedelta(0)
+            else:
+                return self._dst
+
+    tz = TimezoneInfo(utcoffset=timedelta(hours=-10), tzname='US/Hawaii')
+    # The above lines produces a `datetime.tzinfo` object similar to:
+    #     tzinfo = pytz.timezone('US/Hawaii')
+    time = Time('2010-09-03 00:00:00')
+    tz_aware_datetime = time.to_datetime(tz)
+    forced_to_astropy_time = Time(tz_aware_datetime)
+    assert tz.tzname(time.datetime) == tz_aware_datetime.tzname()
+    assert time == forced_to_astropy_time
```

Comment:
It doesn't look like you are checking that `tz_aware_datetime` has the expected local time representation.